runtimerevolution/web3-roulette
# [WEB3-41] TA integration and Admin endpoints

# Description

This PR integrates the connection with team allocator api, to confirm that the user is active and also to get unit information. Additionally, it includes the protection for admin users only in some endpoints, such as creating a new giveaway, and participants management.

Ticket: https://runtime-revolution.atlassian.net/browse/WEB3-41

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

## How Has This Been Tested?

1. Login with your account and verify the user saved in the database (check the unit)
2. Try to create a new giveaway as a normal user and verify that you don't have the right access
3. Verify that it is no longer possible to manage users when the giveaway is not active

## Checklist:
